.github/workflows/build-and-run.yml

```diff
@@ -0,0 +1,128 @@
+name: Build-And-Run Tests
+
+on: [pull_request, push]
```

Comment:
Is the intention that this would run on pushes to *all* branches, or just pushes to `develop`?

I have something like this in the snaphu-py:

```yaml
on:
  pull_request:
  push:
    branches:
      - develop
      - release-*
    tags:
      - v*.*.*
```

So that the CI is triggered only by PRs and pushes to specific branches/tags, rather than pushes to any branch.

Reply:
I was thinking that all our branches here will be develop or release branches, so we'll want to run this workflow on all of them

Reply:
This `ci` branch isn't a develop or release branch! 🤔

<img width="466" alt="image" src="https://github.com/user-attachments/assets/2470a5be-4610-4f1c-b7c1-8fa1f0b2ec15" />

Reply:
Yeah, I think I have to do changes to github action workflows on branches on the main repo, because I thought changes to these files from forks are ignored for security reasons